### DISCUSSION TITLE
Interpolant for vectors

### DISCUSSION THREAD

**Discussion by mzanon** [casadi/casadi#3702] [General]
Hi guys, I am sorry if I ask something that is already well-known, but I would like to know if interpolant can also work with vectors. So far what I understood is that I need to call interpolant separately for each component of my vector, but it would be very practical if I could just pass a vector and obtain back a vector when evaluating the function.

One more thing: somewhere I read about plugin "smooth-linear". I am on casadi 3.6.5 but I get an error if I try to call that. has that been removed? In any case, what did that do? I would be very interested in knowing the details better.

Thanks a lot in advance!

**Comment by jaeandersson** [comment]
For any function object you can make a call with multiple right hand sides. If an argument is supposed to be scalar, you can pass a row vector and all components will be evaluated. If you have a column vector, you need to transpose it. 

@jgillis needs to answer the other question

**Reply by mzanon** [reply]
I am not really sure what that would mean in terms of code. Do you have a minimal example?

**Reply by b-mare** [reply]
Hi, 

You can create your interpolant and supply it with an array.

Minimal example:
```
import casadi as ca
import numpy as np

xgrid = np.linspace(1, 6, 6)
V = [-1, -1, -2, -3, 0, 2]
lut = ca.interpolant("LUT", "bspline", [xgrid], V)
print(lut([2.0, 2.5, 3, 3.5, 4.0, 4.5, 5.0, 5.5, 6.0]))
```

Is this the desired behavior?

**Comment by cdhainaut** [comment]
@mzanon This is how I use the smooth_linear plugin:

```python
interp = interpolant(
    "interpolation",
    "bspline",
    knots,
    values,
    {"algorithm": "smooth_linear", "smooth_linear_frac": 0.4},
)
```

**Reply by mzanon** [reply]
Thank you very much! This is very helpful! 
@jgillis Would be grand if on top of "smooth_linear_frac" there were another parameter to tune when the ramping starts
